workspaces/jupyterlab-python/install.sh

```diff
@@ -2,18 +2,44 @@
 
 set -ex
 
+# This script requires root.
+[ "$(id -u)" -eq 0 ] || exit 1
+
+# On PL, we want to standardize on using 1001:1001 for the user. We change
+# the "jovyan" account's default UID and GID here.
+USER_NAME=jovyan
+GROUP_NAME=$(id -gn $USER_NAME)
+OLD_UID=$(id -u $USER_NAME)
+OLD_GID=$(id -g $USER_NAME)
+NEW_UID=1001
+NEW_GID=1001
+groupmod -g $NEW_GID $GROUP_NAME
+usermod -u $NEW_UID -g $NEW_GID $USER_NAME
+for d in /home /tmp ; do
+    find $d -user $OLD_UID -execdir chown -h $NEW_UID {} +
+    find $d -group $OLD_GID -execdir chgrp -h $NEW_GID {} +
+done
+
 # Install dependencies and various libraries
 apt-get update && apt-get -y upgrade
-apt-get -y install gcc graphviz graphviz-dev
+apt-get -y install gosu gcc graphviz graphviz-dev
+
+# Test gosu and prepare the gosu helper.
+gosu jovyan true || exit 1
+mv /pl-gosu-helper.sh /usr/local/bin/
+chown root:root /usr/local/bin/pl-gosu-helper.sh
+chmod 0755 /usr/local/bin/pl-gosu-helper.sh
 
 # Install all Python dependencies.
-pip3 install -r /requirements.txt
+pip install uv
+uv pip install --system -r /requirements.txt
```

Comment:
If someone wants to build a customized image on top of this one, would this mean they'd have to use `uv` too, or would using `pip` play nicely here (install into the same global namespace without conflicts and such)?

From my experience with conda/mamba in other images, I'd expect things to work like this, but just wanted to check since we haven't used `uv` before.

Reply:
I thought I had submitted these comments already but it looks like they were in draft only. If there are any doubts about `uv` here then by all means, we can just use pip instead.

Reply:
> If someone wants to build a customized image on top of this one, would this mean they'd have to use uv too

If we create a base jupyterlab image without any python packages, then that could be the main target for building custom images and make this point less of a restriction. I believe that would also allow those who want to use the built-in mamba installer to  install `gcc` and `graphviz`, which I assume are needed for some of the python packages and not for any core PL/docker functionality?

Reply:
I went ahead and removed `uv` so that this won't be a blocker any longer.

Reply:
I do think the speed improvements of `uv` are intriguing all around (including perhaps for the PL base image itself), but after thinking about this I'd rather limit the number of things that this PR is changing. I see you reverted to `pip` in a recent comment, I think that's the right move for now.

Reply:
@joelostblom It looks like just installing python3 and python3-pip on Ubuntu pulls in gcc and g++ (among other things). I think the other PL images ran into some trouble getting graphviz installed depending on the distro (for the latest VS Code image, mamba was used to install the binary).